plugins/plugin-bash-like/src/lib/cmds/catchall.ts

```diff
@@ -35,6 +35,10 @@ export const dispatchToShell = async ({
 }: EvaluatorArgs) => {
   debug('handling catchall', command)
 
+  if (!execOptions.exec) {
```

Comment:
i think this should be execOptions.type === ExecType.TopLevel; i.e. reject user's trying to issue `!`.

Reply:
and it's ok not to include this change in this PR.